cmd/build.go

```diff
@@ -312,7 +357,21 @@ func (c buildConfig) Prompt() (buildConfig, error) {
 	if err != nil {
 		return c, err
 	}
-	if (f.Registry == "" && c.Registry == "" && c.Image == "") || c.Confirm {
+
+	// Check if function exists first
+	if !f.Initialized() {
+		// Return a specific error for uninitialized function
+		return c, fn.NewErrNotInitialized(f.Root)
+	}
+
+	// Check if registry/image is missing BEFORE prompting
+	if f.Registry == "" && c.Registry == "" && c.Image == "" {
+		// Return error immediately - don't prompt
+		return c, fn.ErrRegistryRequired
+	}
```

Comment:
I think when a Function is initialized AND registry/image is missing, we can prompt for registry. Because this would effectively make `--registry` a forced flag if you havent specified FUNC_REGISTRY or used the config file.
Something like
```
func init
func deploy  (/build)
``` 
without any prior setup should not fail, rather prompt for the registry since this is the minimal requirement.
I think thats a bit more user friendly